sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -374,6 +418,28 @@ RestClient createClient() throws IOException {
           throw new IOException("Can't load the client certificate from the keystore", e);
         }
       }
+      restClientBuilder.setRequestConfigCallback(
```

Comment:
That seems a bit complicated to me but maybe I'm missing something here.
Why not use something like:
```suggestion
          restClientBuilder.setRequestConfigCallback(
          new RestClientBuilder.RequestConfigCallback() {
            @Override
            public RequestConfig.Builder customizeRequestConfig(
                RequestConfig.Builder requestConfigBuilder) {
              if (getConnectTimeout() != null) {
                requestConfigBuilder.setConnectTimeout(getConnectTimeout());
              }
              if (getSocketAndRetryTimeout() != null){
                requestConfigBuilder.setSocketTimeout(getSocketAndRetryTimeout());
              }
              return requestConfigBuilder;
            }
          });

      if (getSocketAndRetryTimeout() != null) {
        restClientBuilder.setMaxRetryTimeoutMillis(getSocketAndRetryTimeout());
      }

```